### ISSUE TITLE
IDE0040: Accessibility modifiers required should not be reported on file types

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#62259]
Related to #60819

File types cannot specify accessibility modifiers, so the IDE shouldn't suggest doing so.

**Actual behavior**:
```cs
file class FC // info IDE0040: Accessibility modifiers required
{
}
```

**Expected behavior**:
```cs
file class FC // no diagnostic
{
}
```

**Comment by Youssef1313** [comment]
https://github.com/dotnet/roslyn/blob/58d8ea0066f649de1bfe1852bb90f56db981bb67/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpAccessibilityFacts.cs#L30-L44
